### PR TITLE
Add improved queue feature

### LINKED ISSUE #228: Add an improved automatic syncing system.
At the moment you will automatically sync with others when you come to a near stop. I have an issue where I slow down for a moment while racing and driving and it loads the vehicles and causes a huge lag spike during the worst times. This can be quite frustrating. 

My solution would be to have a custom setting for the player to chose how long and at what speed they must be under for it to sync.

An alternative would be to make this an value that is not adjustable and just improve the timing and current syncing a little bit, but I thought it might go well to give the player the option.

For example the player must be under 1 mph (in any direction) and must be consistently below that speed for lets say 2 seconds. I thought this could a great addition to providing a much smoother and cleaner game-play.

### PR DESCRIPTION
**Pull request by 20dka**
This PR incorporates code from @jacobcargen and mine building a UI around it.

### CODE REVIEW

**Review by Olrosse — APPROVED** [review]

**Review by Starystars67 — APPROVED** [review]
Looks good however I have not been able to test, on a code level though it looks good.

**Comment by 20dka** [comment]
this resolves #228

**Comment by 20dka** [comment]
todo: move queue button to avoid accidental server leaving

**Comment by O1LER** [comment]
> todo: move queue button to avoid accidental server leaving

Personally, i would move the quit button, all the way to the left, since quit button and servername are both about the server while event queue has somewhat a connection to players, so itsmore fitting next to the number of players.

Also, is there a keybind to manually load queue? If not, that might be nice too

File: ui/modules/options/multiplayer.partial.html [hunk 1]
```diff
@@ -40,14 +40,37 @@ <h3>{{:: 'ui.options.general' | translate}}</h3>
     <md-checkbox ng-model="options.data.values.enableQueueAuto" ng-change="options.applyState()"></md-checkbox>
   </md-list-item>
 
-  <md-list-item md-no-ink ng-show="options.showAdvancedMPOptions">
+  <md-list-item md-no-ink>
+    <md-tooltip md-direction="right">{{:: 'ui.options.multiplayer.queueApplySpeed.tooltip' | translate }}</md-tooltip>
+    <p flex="50"><span class="mp-setting-highlight">NEW</span> {{:: 'ui.options.multiplayer.queueAutoSkipRemote' | translate }}</p>
```
Comment (Mack29446) [inline ui/modules/options/multiplayer.partial.html:45]: This is referring to the wrong locale

File: ui/modules/options/multiplayer.partial.html [hunk 2]
```diff
@@ -40,14 +40,37 @@ <h3>{{:: 'ui.options.general' | translate}}</h3>
...
+      (options.data.values.queueApplySpeed*2.23694).toFixed(1) }} mph</span>
+    <span ng-if="options.data.values.uiUnitLength != 'imperial'" style="margin-left:15px">{{
+      (options.data.values.queueApplySpeed*3.6).toFixed(1) }} km/h</span>
+    <span style="margin-left:5px">({{ (options.data.values.queueApplySpeed).toFixed(0) }} m/s)</span>
+
+  </md-list-item>
+
+  <md-list-item md-no-ink>
+    <md-tooltip md-direction="right">{{:: 'ui.options.multiplayer.queueApplyTimeout.tooltip' | translate }}</md-tooltip>
+    <p flex="50"><span class="mp-setting-highlight">NEW</span> {{:: 'ui.options.multiplayer.queueAutoSkipRemote' | translate }}</p>
```
Comment (Mack29446) [inline ui/modules/options/multiplayer.partial.html]: This is also using the wrong locale

File: ui/modules/options/multiplayer.partial.html (hunk 1 above)
Comment (Mack29446) [inline ui/modules/options/multiplayer.partial.html]: This is referring to the wrong locale

File: locales/en-US.json [hunk 3]
```diff
@@ -6257,7 +6257,11 @@
     "ui.options.multiplayer.queueSpawnEdit.tooltip": "If enabled, remote vehicle spawns and edits will be put in a queue, instead of applying instantly",
     "ui.options.multiplayer.queueAuto": "Automatic queue system",
     "ui.options.multiplayer.queueAuto.tooltip": "If enabled, you will only spawn queued vehicles once you come to a stop",
-    "ui.options.multiplayer.queueAutoSkipRemote": "Don't queue if spectating others",
+    "ui.options.multiplayer.queueApplySpeed": "Queue apply speed treshold",
```
Comment (Mack29446) [inline locales/en-US.json]: Please fix this typo - it should be `threshold` instead of `treshold`

File: locales/en-US.json [hunk 4]
```diff
@@ -6257,7 +6257,11 @@
...
+    "ui.options.multiplayer.queueApplySpeed.tooltip": "Your vehicle has to be moving slower than this speed for queues to apply",
+    "ui.options.multiplayer.queueApplyTimeout": "Queue apply timeout",
+    "ui.options.multiplayer.queueApplyTimeout.tooltip": "You have to move under your set speed treshold for this long before queues apply",
```
Comment (Mack29446) [inline locales/en-US.json]: Same typo here

**Review by Mack29446 — CHANGES_REQUESTED** [review]

**Review by Mack29446 — APPROVED** [review]
Thank you! LGTM

**Pull request merged**